cmd/common-main.go

```diff
@@ -189,8 +189,8 @@ func minioConfigToConsoleFeatures() {
 	}
 	os.Setenv("CONSOLE_MINIO_REGION", globalSite.Region)
 	os.Setenv("CONSOLE_CERT_PASSWD", env.Get("MINIO_CERT_PASSWD", ""))
-	if globalSubnetConfig.License != "" {
-		os.Setenv("CONSOLE_SUBNET_LICENSE", globalSubnetConfig.License)
+	if globalSubnetConfig.ApiKey != "" {
+		os.Setenv("CONSOLE_SUBNET_API_KEY", globalSubnetConfig.ApiKey)
 	}
 }
```

Comment:
I propose we do not change the terminology of the environment variable here API_KEY is still a LICENSE so we should keep it that way to avoid breaking changes IMO.

We cannot arbitrarily change the envs as this will break operator deployments etc.  Keep the license as is we should simply remove the verification of the keys and the subnet server should handle both JWT and UUID for some time IMO. 

We already have customers using JWT based deployments changing this now can create a lot of problems migrating. 

We should remove the parsing logic such that any opaque key can be added, and keep both for some time and change to API_KEY after another 6months to 1yr down the line.

Reply:
In that case, I propose to retain `CONSOLE_SUBNET_LICENSE` as is, deprecate it, and introduce `CONSOLE_SUBNET_API_KEY` as a new env variable. That way,
 - license will continue to work in existing deployments
 - new deployments can use api key
 - there will be a clear distinction between the two
 - it will be simpler to remove the license part at a later date.
 - currently the license is partly being used like an api key, but I don't think api key can be considered as a `license`, and hence better to have a properly named variable for it.

I think an env variable like this may not be required for console in future, as console will also be using the minio config (`subnet api_key`) to get the api key, similar to what `mc` does.

Reply:
@anjalshireesh correct but API_KEY itself doesn't designate itself as License - however, it is used as a license. i.e it's given out only for paid customers. 

That is why I don't know why it's useful to rename it. API_KEY sounds like CLIENT_ID in OpenID which has a completely different meaning.

LICENSE here gives an indication to customers that it is special i.e

- it is used to designate (licensed setup)
- license to operate the cluster (removing AGPL exceptions)

API_KEY, however, is meaningful for communicating to subnet - it is not meant for end-user and they may still have to mentally imagine that API_KEY is the license key.

So my suggestion is LICENSE has a better meaning than API_KEY for end-users.

Reply:
@harshavardhana I just don't like the idea of the same config variable being used for storing two different kind of things, having different formats altogether. The two might be similar at an abstract level, but are still very much different. This will also force the clients (mc/console) to have extra logic to understand what it is and accordingly pass appropriate param to subnet. (subnet api expects different query params for license/api-key).

So I'm now suggesting NOT to replace existing license config, but to introduce a new config/env for api key. New versions of mc will always deal with api key, older ones can continue to use license.